models/trucks/trucks_query.go

```diff
@@ -0,0 +1,67 @@
+package models
+
+import (
+	"errors"
+	trips "kargo-back/models/trips"
+	"strconv"
+)
+
+var (
+	// ErrInvalidOrigin error when an invalid origin is given
+	ErrInvalidOrigin = errors.New("invalid origin parameter")
+	// ErrInvalidDestination error when an invalid destination is given
+	ErrInvalidDestination = errors.New("invalid destination parameter")
+	// ErrInvalidVolume error when truck max volume is invalid
+	ErrInvalidVolume = errors.New("invalid volume parameter")
+	// ErrInvalidWeight error when truck max weight is invalid
+	ErrInvalidWeight = errors.New("invalid weight parameter")
+)
+
+type TrucksQuery struct {
+	TruckType   string
+	Origin      string
+	Destination string
+	Weight      float64
+	Volume      float64
+}
+
+func convertToValidFloat(floatString string) (float64, bool) {
+	float, err := strconv.ParseFloat(floatString, 64)
+	if err != nil {
+		return 0, false
+	}
+
+	return float, float >= 0
+}
```

Comment:
This could be moved to shared as is not directly tied to query trucks but is more of a general function that later can (nah, will) be used somewhere else.